corehq/apps/app_manager/static/app_manager/js/forms/advanced/actions.js

```diff
@@ -63,7 +63,9 @@ hqDefine('app_manager/js/forms/advanced/actions', function() {
                 if (!self.auto_select && !tag) {
                     // Don't allow user to blank out case tag
                     self.warn_blank_case_tag(true);
-                    self.case_tag(self.case_tag.previous());
+                    if(hqImport('hqwebapp/js/toggles').toggleEnabled('ALLOW_BLANK_CASE_TAGS')) {
```

Comment:
I think this is also backwards (assuming the default is to preserve the old behavior).
Also, what does this part do?